homeassistant/components/ialarm/config_flow.py

```diff
@@ -0,0 +1,66 @@
+"""Config flow for Antifurto365 iAlarm integration."""
+import logging
+
+from pyialarm import IAlarm
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_HOST, CONF_PORT
+
+# pylint: disable=unused-import
+from .const import DEFAULT_PORT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+    }
+)
+
+
+async def _get_device_mac(hass: core.HomeAssistant, host, port):
+    ialarm = IAlarm(host, port)
+    mac = await hass.async_add_executor_job(ialarm.get_mac)
+    return mac
+
+
+class IAlarmConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Antifurto365 iAlarm."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        mac = None
+
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
+
+        host = user_input[CONF_HOST]
+        port = user_input[CONF_PORT]
+
+        try:
+            # If we are able to get the MAC address, we are able to establish
+            # a connection to the device.
+            mac = await _get_device_mac(self.hass, host, port)
+            if mac is None:
+                errors["base"] = "cannot_connect"
```

Comment:
Can this happen without it throwing?

Reply:
I'm not 100% sure about that. I guess that there is the remote possibility that the alarm sends a malformed but valid reply that doesn't contain the MAC address that we asked. In that case `get_mac()` will return `None` without throwing.

I don't know if this is something that might happen in the real world, but probably it's better be safe and check for `None`?

Reply:
Ideally if there is malformed data coming back from the device there is an exception thrown by the underlying library

Reply:
Right, I tagged a new release of the pyialarm library. Now if it doesn't receive the expected parameter from the alarm, a ConnectionError will be raised.
For this reason I removed the `if mac is None` check here.